app/models/base_profile.rb

```diff
@@ -7,10 +7,10 @@ class BaseProfile < ApplicationRecord
   default_scope -> { kept }
 
   after_discard do
-    user.discard!
+    user.discard! unless user.discarded?
   end
 
   after_undiscard do
-    user.undiscard!
+    user.undiscard! unless user.kept?
```

Comment:
Does this differ from `(un)discard` without the exclamation mark?

Reply:
Sorted